### ISSUE TITLE
Cannot import flax with v0.7.3

### ISSUE THREAD

**Issue by sanchit-gandhi** [google/flax#3329]
### System information
- OS Platform and Distribution: Linux-5.13.0-1023-gcp-x86_64-with-glibc2.29
- Flax, jax, jaxlib versions: 0.7.3, 0.4.13, 0.4.13 
- Python version: 3.8.10  
- GPU/TPU model and memory: TPU v4-8
- CUDA version (if applicable): NA

### Problem you have encountered:
Cannot import Flax.

### What you expected to happen:
Able to import Flax.

### Logs, error messages, etc:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/sanchitgandhi/hf/lib/python3.8/site-packages/flax/__init__.py", line 25, in <module>
    from . import linen
  File "/home/sanchitgandhi/hf/lib/python3.8/site-packages/flax/linen/__init__.py", line 34, in <module>
    from .activation import (
  File "/home/sanchitgandhi/hf/lib/python3.8/site-packages/flax/linen/activation.py", line 21, in <module>
    from flax.linen.module import compact
  File "/home/sanchitgandhi/hf/lib/python3.8/site-packages/flax/linen/module.py", line 68, in <module>
    from flax.linen import kw_only_dataclasses
  File "/home/sanchitgandhi/hf/lib/python3.8/site-packages/flax/linen/kw_only_dataclasses.py", line 126, in <module>
    def _process_class(cls: type[M], extra_fields=None, **kwargs):
TypeError: 'type' object is not subscriptable
```

### Steps to reproduce:
```bash
python -c "import flax" 
```

### Further info:
Am able to import Flax up to v0.7.2, but not higher.

### Who can help:

@cgarciae

**Comment by cgarciae** [comment]
Hey @sanchit-gandhi ! Can you upgrade to Python 3.9? We dropped support for python 3.8 recently following JAX / numpy.

Else, please downgrade to `flax==0.7.2`.

**Comment by sanchit-gandhi** [comment]
Hey @cgarciae - thanks for the prompt reply! I'll ping `flax` to 0.7.2 as the 'quick fix' and start from python 3.9+ in future instances, thanks!
